cli/generate.php

```diff
@@ -77,20 +83,89 @@
 
 $recipefile = $options['recipe'];
 
-// Expanding '~' on Unix-like OS'es.
-if ($recipefile[0] === '~') {
-    $homedir = getenv('HOME');
-    $recipefile = $homedir.substr($recipefile, 1);
-}
-
-$recipefile = realpath($recipefile);
+$recipefile = expand_path($recipefile);
 if ($recipefile === false) {
     echo("\nInvalid recipe file!\n");
     echo $help;
     die();
 }
 
-$loglevels = Logger::getLevels();
+if (!is_readable($recipefile)) {
+    echo("\nRecipe file not readable!\n");
+    echo $help;
+    die();
+}
+
+// Load the recipe from file.
+$recipe = yaml::decode_file($recipefile);
+
+if (!empty($options['target-dir']) && !empty($options['target-moodle'])) {
+    echo("\nSpecify one of 'target-dir' or 'target-moodle'!\n");
+    echo $help;
+    die();
+}
+
+if (!empty($options['target-dir'])) {
+
+    $targetdir = $options['target-dir'];
+    $targetdir = expand_path($targetdir);
+    if ($targetdir === false) {
+        echo("\nInvalid target directory!\n");
+        echo $help;
+        die();
+    }
+
+    if (!is_writable($targetdir)) {
+        echo("\nTarget plugin location is not writable!\n");
+        echo $help;
+        die();
+    }
+
+    list($type, $name) = \core_component::normalize_component($recipe['component']);
```

Comment:
I think we should perform the same check here as in https://github.com/mudrd8mz/moodle-tool_pluginskel/blob/df78114125a89dfe38b7811865afa4de02461090/classes/local/util/manager.php#L272